pkg/compactor/blocks_cleaner.go

```diff
@@ -410,15 +424,39 @@ func (c *BlocksCleaner) cleanUser(ctx context.Context, userID string, userLogger
 		}
 	}()
 
-	// Read the bucket index.
+	retries := backoff.New(ctx, backoff.Config{
+		MaxRetries: 5,
```

Comment:
5 retries seems a lot, especially if the system is timing out because it's overloaded, we're just making it worse. Maybe 2-3?